### ISSUE TITLE
Abrir Code Tray ao iniciar o sistema

### ISSUE THREAD

**Issue by luizjr** [diego3g/youtube-challenge-electron-tray#13]
Gostei bastante de acompanhar esse desafio, eu acredito que usaria no meu dia a dia.
Mas um detalhe importante seria ele abrir sempre que o Mac iniciar.

**Comment by diego3g** [comment]
Fala @luizjr, parece que existe um pacote pra isso, o `auto-launch` (https://www.npmjs.com/package/auto-launch). Tens vontade de contribuir com o projeto adicionando isso? Seria legal o usuário ter uma opção pra escolher se ele quer ou não que isso aconteça, o que acha?

**Comment by luizjr** [comment]
> Fala @luizjr, parece que existe um pacote pra isso, o `auto-launch` (https://www.npmjs.com/package/auto-launch). Tens vontade de contribuir com o projeto adicionando isso? Seria legal o usuário ter uma opção pra escolher se ele quer ou não que isso aconteça, o que acha?

Eu ainda estou aprendendo, mas quero sim tentar ajudar.
Vou tentar implementar esse pacote na aplicação ainda hoje.

**Comment by rodrigocananea** [comment]
> Fala @luizjr, parece que existe um pacote pra isso, o `auto-launch` (https://www.npmjs.com/package/auto-launch). Tens vontade de contribuir com o projeto adicionando isso? Seria legal o usuário ter uma opção pra escolher se ele quer ou não que isso aconteça, o que acha?

O próprio Electron.js já tem uma solução para isso
```js
app.setLoginItemSettings({
  openAtLogin: true,
  path: app.getPath('exe'),
  args: [
    '--processStart', `"${exeName}"`,
    '--process-start-args', '"--hidden"',
  ],
})`

**Comment by luizjr** [comment]
> > Fala @luizjr, parece que existe um pacote pra isso, o `auto-launch` (https://www.npmjs.com/package/auto-launch). Tens vontade de contribuir com o projeto adicionando isso? Seria legal o usuário ter uma opção pra escolher se ele quer ou não que isso aconteça, o que acha?
> 
> O próprio Electron.js já tem uma solução para isso
> 
> ```js
> app.setLoginItemSettings({
>   openAtLogin: true,
>   path: app.getPath('exe'),
>   args: [
>     '--processStart', `"${exeName}"`,
>     '--process-start-args', '"--hidden"',
>   ],
> })`
> ```

Eu fiz com aquele pacote que o @diego3g recomendou.
Fiquei com medo da solução que você citou por ver a extensão ".exe" no path.

Você já testou?

Você pode conferir também o que eu fiz nas solicitações de pull: #16

**Comment by rodrigocananea** [comment]
> > > Fala @luizjr, parece que existe um pacote pra isso, o `auto-launch` (https://www.npmjs.com/package/auto-launch). Tens vontade de contribuir com o projeto adicionando isso? Seria legal o usuário ter uma opção pra escolher se ele quer ou não que isso aconteça, o que acha?
> > 
> > 
> > O próprio Electron.js já tem uma solução para isso
> > ```js
> > app.setLoginItemSettings({
> >   openAtLogin: true,
> >   path: app.getPath('exe'),
> >   args: [
> >     '--processStart', `"${exeName}"`,
> >     '--process-start-args', '"--hidden"',
> >   ],
> > })`
> > ```
> 
> Eu fiz com aquele pacote que o @diego3g recomendou.
> Fiquei com medo da solução que você citou por ver a extensão ".exe" no path.
> 
> Você já testou?
> 
> Você pode conferir também o que eu fiz nas solicitações de pull: #16

Sim, havia testado isso, mas como mencionou o "exe" ali, eu testei somente no Windows, dai vendo qual seria a melhor abordagem seria até interessante utilizar envirioment do electronjs pra controlar como ele deveria encontrar o app para abrir. Ou utilizar o pacote pra uma solução mais rápida hehe

**Comment by luizjr** [comment]
> > > > Fala @luizjr, parece que existe um pacote pra isso, o `auto-launch` (https://www.npmjs.com/package/auto-launch). Tens vontade de contribuir com o projeto adicionando isso? Seria legal o usuário ter uma opção pra escolher se ele quer ou não que isso aconteça, o que acha?
> > > 
> > > 
> > > O próprio Electron.js já tem uma solução para isso
> > > ```js
> > > app.setLoginItemSettings({
> > >   openAtLogin: true,
> > >   path: app.getPath('exe'),
> > >   args: [
> > >     '--processStart', `"${exeName}"`,
> > >     '--process-start-args', '"--hidden"',
> > >   ],
> > > })`
> > > ```
> > 
> > 
> > Eu fiz com aquele pacote que o @diego3g recomendou.
> > Fiquei com medo da solução que você citou por ver a extensão ".exe" no path.
> > Você já testou?
> > Você pode conferir também o que eu fiz nas solicitações de pull: #16
> 
> Sim, havia testado isso, mas como mencionou o "exe" ali, eu testei somente no Windows, dai vendo qual seria a melhor abordagem seria até interessante utilizar envirioment do electronjs pra controlar como ele deveria encontrar o app para abrir. Ou utilizar o pacote pra uma solução mais rápida hehe

Concordo Contigo, essa ideia será usada quando eu for evoluir o Launcher.

**Comment by cymonbr** [comment]
Eu criei esse app com uma abordagem diferente por isso acabei montando fora daqui, mas vou ver se implemento isso no meu, estava na procura de alguma forma de fazer ele iniciar com o sistema. Valeu
